applications/asset_tracker_v2/src/modules/modules_common.h

```diff
@@ -7,35 +7,41 @@
 #ifndef _MODULES_COMMON_H_
 #define _MODULES_COMMON_H_
 
+#include <zephyr.h>
+
 #define IS_EVENT(_ptr, _mod, _evt) \
-		is_ ## _mod ## _module_event(&_ptr->module._mod.header) &&     \
+		is_ ## _mod ## _module_event(&_ptr->module._mod.header) &&		\
 		_ptr->module._mod.type == _evt
 
-#define SEND_EVENT(_mod, _type)						       \
-	struct _mod ## _module_event *event = new_ ## _mod ## _module_event(); \
-	event->type = _type;						       \
+#define SEND_EVENT(_mod, _type)								\
+	struct _mod ## _module_event *event = new_ ## _mod ## _module_event();		\
+	event->type = _type;								\
+	EVENT_SUBMIT(event)
+
+#define SEND_ERROR(_mod, _type, _error_code)						\
+	struct _mod ## _module_event *event = new_ ## _mod ## _module_event();		\
+	event->type = _type;								\
+	event->data.err = _error_code;							\
 	EVENT_SUBMIT(event)
 
-#define SEND_ERROR(_mod, _type, _error_code)				      \
-	struct _mod ## _module_event *event = new_ ## _mod ## _module_event(); \
-	event->type = _type;						      \
-	event->data.err = _error_code;					      \
+#define SEND_SHUTDOWN_ACK(_mod, _type, _id)						\
+	struct _mod ## _module_event *event = new_ ## _mod ## _module_event();		\
+	event->type = _type;								\
+	event->data.id = _id;								\
 	EVENT_SUBMIT(event)
 
 struct module_data {
+	/* Variable used to construct a linked list of module metadata. */
+	sys_snode_t header;
+	/* ID specific to each module. Internally assigned when calling module_start(). */
+	uint32_t id;
 	k_tid_t thread_id;
 	char *name;
 	struct k_msgq *msg_q;
+	/* Flag signifying if the module supports shutdown. */
+	bool supports_shutdown;
```

Comment:
Note for later improvement: We can add a `has_thread` flag here. Then we can use the flag to populate the `thread_id` member in `module_start`

Reply:
Agreed. Added to backlog.